### PR TITLE
added saving scheduler and optimizer

### PR DESCRIPTION
**Pull request by kategimranova**

### CODE REVIEW

File: src/utils/train.py [hunk 1]
```diff
@@ -139,7 +140,15 @@ def train(
...
                 )
             if batch_num % config.save_every == 0:
-                torch.save(model.state_dict(), os.path.join(validation_wandb_path, f"batch_{batch_num}.ckpt"))
+                torch.save(
+                    {
+                        "scheduler": scheduler.state_dict(),
+                        "model": model.state_dict(),
+                        "optimizer": optimizer.state_dict(),
+                        "loss": loss,
+                    },
```
Comment (SpirinEgor) [inline src/utils/train.py:149]: That's very useful, please also add loading these states on the model checkpoint load

Comment (kategimranova) [reply src/utils/train.py:149]: what model checkpoint load do you mean?

**Review by SpirinEgor — APPROVED** [review]
